src/components/habit-tracker/analytics/month-view/LineChart.tsx

```diff
@@ -0,0 +1,121 @@
+import ReactApexChart from "react-apexcharts";
+import styled from "styled-components";
+import { Logging } from "../../../../containers/habit-tracker/habits/Habits";
+import { colors } from "../../../../common/utilities/color-utils";
+
+// Styled component for the chart container
+const ChartContainer = styled.div`
+  width: 100%;
+  height: calc(100% - 40px);
+  margin-top: 40px;
+
+  .apexcharts-canvas {
+    background: ${colors.white};
+    border-radius: 8px;
+    box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);
+  }
+
+  @media (max-width: 768px) {
+    height: 70vh;
+  }
+
+  @media (max-width: 480px) {
+    height: 50vh;
+  }
+`;
+
+// interface for the line chart props
+interface LineChartProps {
+  habitsLog: Logging;
+  selectedHabit: string;
+  year: number;
+  month: string;
+}
+
+// functional component to render the line chart for habits logged in a month
+export function LineChart({
+  habitsLog,
+  selectedHabit,
+  year,
+  month,
+}: LineChartProps) {
+  const logsForHabit = habitsLog[selectedHabit]?.[year]?.[month] || {};
+  const daysInMonth = new Date(year, new Date().getMonth() + 1, 0).getDate();
+
+  const seriesData = Array.from(
+    { length: daysInMonth },
+    (_, i) => logsForHabit[i + 1] || 0
+  );
+
+  const options: ApexCharts.ApexOptions = {
```

Comment:
define `options` outside the LineChart component, to avoid re-creating this every time LineChart is rendered.